index.js

```diff
@@ -1,29 +1,62 @@
-/* globals require, __dirname, process */
-
 const path = require('path');
+const builder = require('adguard-hostlists-builder');
 const fs = require('fs');
-const builder = require("adguard-hostlists-builder");
-
-// Validate services.json and make sure it is a valid JSON.
-try {
-    JSON.parse(fs.readFileSync('assets/services.json', 'utf8'));
-} catch (ex) {
-    console.error('Failed to parse services.json', ex);
-    process.exit(1);
-}
-
-// Compile hostlists.
+const { checkRemovedServices } = require('./scripts/services/check-removed-services');
+const { rewriteServicesJSON } = require('./scripts/services/rewrite-services-json');
 
 const filtersDir = path.join(__dirname, './filters');
 const assetsDir = path.join(__dirname, './assets');
 const tagsDir = path.join(__dirname, './tags');
 const localesDir = path.join(__dirname, './locales');
+const inputServicesDir = path.join(__dirname, './services');
+const outputServicesFile = path.join(assetsDir, 'services.json');
 
+/**
+ * Validate services.json and make sure it is a valid JSON.
+ *
+ * @param {string} filePath The file path for the "services.json" file.
+ * @throws {Error} If JSON file is not valid.
+ */
+const validateJson = (filePath) => {
+    try {
+        JSON.parse(fs.readFileSync(filePath, 'utf8'));
+    } catch (error) {
+        console.error('Failed to parse services.json', error);
+        process.exit(1);
+    }
+};
+
+/**
+ * Builds the "services.json" file by performing the following steps:
+ * 1. Check if the services.json file is valid.
+ * 2. Check if the services in the "/services" folder have been deleted by comparing with the data in "services.json".
+ * 3. If the information has been deleted, write the missing files.
+ * 4. Collect information from the services files, sort and overwrite "services.json".
+ *
+ * @param {string} dirPath - The directory path where the services data is located.
+ * @param {string} filePath - The file path for the "services.json" file.
+ * @returns {Promise<void>} A promise that resolves when the building process is complete.
+ */
+const buildServices = async (dirPath, filePath) => {
```

Comment:
neither parameter names —`dirPath`, `filePath`— nor their jsdoc description say nothing about their the purpose. I suggest to change these names:
dirPath -> inputDirPath
filePath -> resultFilePath

Reply:
the same is relevant for checkRemovedServices() and rewriteServicesJSON()

please note that for the parameters (which are currently named `dirPath` and `filePath` in `buildServices()`) have different names inside those two other functions:
```
const buildServices = async (dirPath, filePath) => { ... };

const checkRemovedServices = async (servicesFolderPath, jsonFilePath) => { ... };

const rewriteServicesJSON = async (servicesDir, servicesJSON) => { ... };
```
it makes the code more complicated, and there are some duplications which could be avoided if variable names are unified,
for example, reading of the input yaml-files directory can be done just once instead of twice:
https://uploads.adguard.org/slbdzdrv0p8f5.png

Reply:
should I export the `getYmlFileNames` to rewrite-services-json?

Reply:
> should I change all variable names

all of them contain the same data value and has the same type, so yes

- - -

> should I export the getYmlFileNames to rewrite-services-json or create a separate helper dir for this?

neither of this. getYmlFileNames() function is small enough to have it in index.js in the root dir: 
```
const getYmlFileNames = async (servicesFolderPath) => {
    const ymlFiles = await fs.readdir(servicesFolderPath);
    const ymlFileNames = ymlFiles.map((ymlFile) => path.parse(ymlFile).name);
    const formattedServiceNames = ymlFileNames.map(normalizeFileName);
    return formattedServiceNames.sort();
};
```

you can read the yaml file names in `buildServices()` after `validateJson(filePath);`,
and pass them to `checkRemovedServices()` and `rewriteServicesJSON()` as a parameter after that